kinto_http/batch.py

```diff
@@ -77,14 +95,26 @@ def send(self):
                 logger.debug("\nBatch #{}: \n\tRequest: {}\n\tResponse: {}\n".format(
                     id_request, json.dumps(chunk[i]), json.dumps(response)))
 
-                # Raise in case of a 500
-                if status_code >= 500:
-                    raise exception
+                if not (200 <= status_code < 400):
+                    # One of the server response is an error.
+                    message = '{0} - {1}'.format(status_code, response['body'])
+                    exception = KintoException(message)
+                    exception.request = RequestDict(chunk[i])
+                    exception.response = ResponseDict(response)
+                    # Should we ignore 4XX errors?
+                    raise_on_4xx = status_code >= 400 and not self._ignore_4xx_errors
+                    if raise_on_4xx:
+                        _exceptions.append(exception)
+                    if status_code >= 500:
+                        raise exception
 
                 id_request += 1
 
             self._results.append((resp, headers))
 
+        if _exceptions:
+            raise KintoBatchException(_exceptions)
```

Comment:
Now that I think of it, we could also provide the results, otherwise there is no way to obtain the response within the batch that succeeded :/

Reply:
Yes it makes total sense.